app/src/main/java/com/iic/shopingo/services/UserConnector.java

```diff
@@ -124,6 +99,38 @@ public void setCurrentUser(User currentUser) {
     editor.apply();
   }
 
+  void fetchUserDetails(Session session, final Task<User>.TaskCompletionSource taskCompletionSource) {
```

Comment:
Suggestion: maybe we should name async operations `start*` (`startFetchUserDetails` in this case)?